ledger/complete/ledger.go

```diff
@@ -363,61 +363,66 @@ func (l *Ledger) ExportCheckpointAt(
 			fmt.Errorf("failed to clean up tries to reduce memory usage: %w", err)
 	}
 
-	// TODO enable validity check of trie
-	// only check validity of the trie we are interested in
-	// l.logger.Info().Msg("Checking validity of the trie at the given state...")
-	// if !t.IsAValidTrie() {
-	//	 return nil, fmt.Errorf("trie is not valid: %w", err)
-	// }
-	// l.logger.Info().Msg("Trie is valid.")
-
-	// get all payloads
-	payloads := t.AllPayloads()
-	payloadSize := len(payloads)
-
-	// migrate payloads
-	for i, migrate := range migrations {
-		l.logger.Info().Msgf("migration %d/%d is underway", i, len(migrations))
-
-		start := time.Now()
-		payloads, err = migrate(payloads)
-		elapsed := time.Since(start)
-
-		if err != nil {
-			return ledger.State(hash.DummyHash), fmt.Errorf("error applying migration (%d): %w", i, err)
+	var payloads []ledger.Payload
+	var newTrie *trie.MTrie
+
+	noMigration := len(migrations) == 0
+
+	if noMigration {
+		// when there is no migration, reuse the trie without rebuilding it
+		newTrie = t
+		// when there is no migration, we don't generate the payloads here until later running the
+		// postCheckpointReporters, because the ExportReporter is currently the only
+		// preCheckpointReporters, which doesn't use the payloads.
+	} else {
+		// get all payloads
+		payloads = t.AllPayloads()
+		payloadSize := len(payloads)
```

Comment:
The diff here is a bit confusing, but there is actually no code change from this line (L380) to L425